File: app/src/main/java/com/hr/kurtovic/tomislav/familyboard/models/Message.java
package com.hr.kurtovic.tomislav.familyboard.models;

import com.google.firebase.firestore.ServerTimestamp;

import java.util.Date;

public class Message {

    private String ID;
    private String message;
    private Date dateCreated;
    private User userSender;
    private String urlImage;
    private boolean isAccepted = false;

    public Message() {
    }

    public Message(String message, User userSender) {
        this.message = message;
        this.userSender = userSender;

    }

    public Message(String message, User userSender, String urlImage) {
        this(message, userSender);
        this.urlImage = urlImage;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    @ServerTimestamp
    public Date getDateCreated() {
        return dateCreated;
    }

    public void setDateCreated(Date dateCreated) {
        this.dateCreated = dateCreated;
    }

    public User getUserSender() {
        return userSender;
    }

    public void setUserSender(User userSender) {
        this.userSender = userSender;
    }

    public boolean isAccepted() {
        return isAccepted;
    }

    public void setAccepted(boolean accepted) {
        isAccepted = accepted;
    }

    public String getUrlImage() {
        return urlImage;
    }

    public void setUrlImage(String urlImage){
        this.urlImage = urlImage;
    }

    public String getID() {
        return ID;
    }

    public void setID(String ID) {
        this.ID = ID;
    }
}

File: app/src/main/java/com/hr/kurtovic/tomislav/familyboard/models/Message.kt
package com.hr.kurtovic.tomislav.familyboard.models

import com.google.firebase.firestore.DocumentId
import com.google.firebase.firestore.DocumentReference
import com.google.firebase.firestore.ServerTimestamp
import java.util.*

data class Message(
    @DocumentId
    var id: String? = null,
    var category: String? = null,
    @get:ServerTimestamp
    var dateCreated: Date? = null,
    var memberSenderRef: DocumentReference? = null,
    var content: Map<String, String>? = null,
    var accepted: Boolean = false,
    var memberWhoAcceptedId: String? = null
)